app/views/lookup_keys/_fields.html.erb

```diff
@@ -17,16 +17,28 @@
                    :help_inline => popover('', _('Whether the class parameter value is managed by Foreman.'))
         ) if is_param %>
     <%= param_type_selector(f, :onchange => 'keyTypeChange(this)') %>
-    <%= textarea_f f, :default_value, :value => f.object.default_value_before_type_cast, :size => "col-md-8",
-                   :disabled => (is_param && (!f.object.override || f.object.use_puppet_default)),
-                   :fullscreen => :true,
-                   :rows => :auto,
-                   :help_inline => popover('', _("Value to use when there is no match.")),
-                   :class => "no-stretch" %>
-    <%= checkbox_f(f, :use_puppet_default, :label => _('Use Puppet default'), :size => "col-md-8",
+    <% if f.object.hidden_value? %>
+        <%= password_f f, :default_value, :size => "col-md-8", :no_label => false, :value => f.object.default_value_before_type_cast,
+                       :disabled => (is_param && (!f.object.override || f.object.use_puppet_default)), :fullscreen => true %>
```

Comment:
This is for consistency of the layout when flipping between hidden and not-hidden, and also to save the need to add and remove the fullscreen button when doing so.